classes/01class/exercises/c01-aws01/nimmi89/README.md

```diff
@@ -0,0 +1,440 @@
+# C01-AWS01
+
+## Commands Execution Output
+
+- Commands to create the first EC2 instance and any additional resource required:
+```
+>>>Command to create a key-pair to ssh into the EC2 instance; [--query option and the --output text option to pipe your private key directly into a file with the .pem extension]
+
+aws ec2 create-key-pair \
+ --key-name C01-AWS01KeyPair\
+ --query 'KeyMaterial'\
+ --output text > C01-AWS01KeyPair.pem
+
+>>>Command to Create a security group in your VPC, and add a rule that allows SSH access from anywhere.
+aws ec2 create-security-group \
+--group-name C01-AWS01SSHAccess \
+--description "Security group for SSH access"
+
+>The command returns:
+{
+    "GroupId": "sg-0619b1745d16f5cb6"
+}
+
+>>>Command to add a rule in that security group that allows SSH access from anywhere.
+aws ec2 authorize-security-group-ingress \
+--group-id sg-0619b1745d16f5cb6 \
+--protocol tcp --port 22 --cidr 0.0.0.0/0
```

Comment:
This works, but it's not a good practice to have a rule in your SG with the 0.0.0.0/0 as source(unless this is s public service), specially in a public subnet.
Can you think in a way to make this more secure?

Reply:
Yes, especially not at all a good practice in production code. Have authorized the security group to allow SSH from specific IP. Have made the changes